eng/pipelines/templates/steps/sync-directory.yml

```diff
@@ -15,14 +16,28 @@ steps:
 
   - ${{ each repo in parameters.Repos }}:
     - pwsh: |
+        Write-Host "Patch 1 $(PatchFilesLocation)"
         git clone --branch ${{ parameters.BaseBranchName }} https://github.com/azure/${{ repo }}
         $repoPath = "${{ repo }}/${{ parameters.DirectoryToSync }}"
-
-        if (!(Test-Path $repoPath)) { mkdir $repoPath }
-        Remove-Item -v -r $repoPath
-        Copy-Item -v -r $(Build.SourcesDirectory)/${{ parameters.DirectoryToSync }} $repoPath
-        Get-ChildItem -r $repoPath
-
+        if ('$(PatchFilesLocation)')
+        {
+          pushd ${{ repo }}
+          $results = Get-ChildItem $(PatchFilesLocation) | % { git -c user.name="azure-sdk" -c user.email="azuresdk@microsoft.com" am $_.FullName }
+          if (($results.Count -ne (Get-ChildItem $(PatchFilesLocation)).Count) -or ($LASTEXITCODE -ne 0))
+          {
+            Write-Output "Failed to apply patch files to [https://github.com/azure/${{ repo }}]"
+            exit 1
+          }
+          echo "##vso[task.setvariable variable=HasChanges]$true"
+          echo "##vso[task.setvariable variable=RepoNameWithoutOwner]$repo"
```

Comment:
This is a side-effect of me putting the RepoNameWithoutOwner as part of the same step as the check for changes. We should probably separate that out so we don't have to do this here.